### PR TITLE
Add connect.field_info module & make flattened response from `connect.item_info` configurable

### LINKED ISSUE #34: unexpected results for item_info with field parameter
Hello,

I noticed some unexpected behaviour while using the field argument on the item_info module

Here is an example:

image

1. Query for the field 'web-username'
instead of getting the query result of the field 'web-username' in the op_item dict like { value: 'test-username-web', ... } its in the 'field' attribute - not sure if this is intended 

  tasks:
    - name: 1-Query web-username
      item_info:
        item: TEST-App-1
        vault: my-vault-id
        field: web-username
      no_log: true
      register: debug_one

    - name: Print the result
      ansible.builtin.debug:
        var: debug_one

TASK [Print the result] **********************************************************************************************************************************************************************************************
ok: [localhost] => {
    "debug_one": {
        "changed": false,
        "failed": false,
        "field": "test-username-web",
        "op_item": {}
    }
}

2. Query for a field that exists multiple times within different sections -> only one match is returned, again value is in the field attribute

3. Query the whole item without field, only one username/password is return in the result array - from different sections, i guess this happens during the api -> python transformation where the keys are overwritten.

 tasks:
    - name: 3-Query item
      item_info:
        item: TEST-App-1
        vault: my-vault-id
      no_log: true
      register: debug_item

    - name: Print the result
      ansible.builtin.debug:
        var: debug_item
       
ok: [localhost] => {
    "debug_item": {
        "changed": false,
        "failed": false,
        "op_item": {
            "category": "LOGIN",
            "createdAt": "2021-08-06T17:05:39Z",
            "fields": {
                "notesPlain": {
                    "id": "notesPlain",
                    "label": "notesPlain",
                    "purpose": "NOTES",
                    "type": "STRING"
                },
                "password": {
                    "id": "91F5FD24A0....",
                    "label": "password",
                    "section": {
                        "id": "Section_5A00EF772...."
                    },
                    "type": "CONCEALED",
                    "value": "test-password-web"
                },
                "username": {
                    "id": "5135F336F4....",
                    "label": "username",
                    "section": {
                        "id": "Section_1E8738FADA...."
                    },
                    "type": "STRING",
                    "value": "test-username-haproxy"
                },
                "web-username": {
                    "id": "889126C76....",
                    "label": "web-username",
                    "section": {
                        "id": "Section_5A00EF772966...."
                    },
                    "type": "STRING",
                    "value": "test-username-web"
                }
            },
            "id": "fpdo7wrajn....",
            "lastEditedBy": "FKJ36NA3CVC....",
            "sections": [
                {
                    "id": "linked items",
                    "label": "Related Items"
                },
                {
                    "id": "Section_5A00EF772....",
                    "label": "Web interface"
                },
                {
                    "id": "Section_1E8738FADA.....",
                    "label": "haproxy"
                },
                {
                    "id": "Section_5A00EF772966.....",
                    "label": "mysql"
                }
            ],
            "title": "TEST-App-1",
            "updatedAt": "2021-08-06T17:16:14Z",
            "vault": {
                "id": "my-vault-id"
            },
            "version": 4
        }
    }
}

Furthermore specifying a section for field lookups (to get a specific 'in-item-namespace') is currently not possible. This is a blocker to migrate our main workloads to the 1password connect server since our old entries are organised by sections.

We would be happy to commit a pull request regarding this issues, but we are currently not sure how the intended behaviour  for this module should be. I guess returning all results in op_item as array for field searches, or ignore results from sections and add a dedicated section parameter would be possible. 

Best Klaus

### PR DESCRIPTION
**Pull request by verkaufer**
# Summary

* Adds `connect.field_info` module.
* Introduce `flatten_fields_by_label` toggle to `connect.item_info` module (fixes #34)

## Flattening fields in item_info

Issue #34 raised the problem with non-unique field labels overwriting the field values for the returned item if the field label was already in the dictionary. 

To avoid a breaking API change, this change introduces the `flatten_fields_by_label` option, which defaults to `true` and preserves current behavior.

If the `flatten_fields_by_label` is false, then the `fields` key in the returned `{"op_item: { }"` dict is a list of field data.

Example
```yaml
---
# Omitting `flatten_fields_by_label` => the `fields` key is a dict, non-unique field labels may overwrite each other
- name: Find and return Item details for "Dev-Database" 
  onepassword.connect.item_info:
    item: Dev-Database
    vault: 2zbeu4smcibizsuxmyvhdh57b6

# Set `flatten_fields_by_label` to false => fields are returned as a list, duplicate labels not removed
- name: Find and return Item details for "Dev-Database" - not flattened
  onepassword.connect.item_info:
    item: Dev-Database
    vault: 2zbeu4smcibizsuxmyvhdh57b6
    flatten_fields_by_label: false
```

**Returning the fields as a list will be the default behavior in v3.0.0**

## connect.field_info
You can now have your playbooks search for a specific field within an item. 

The module accepts the following options:
* field - the field label or its UUID
* item - item name or its UUID
* vault - the vault UUID 

Example:
```yaml
- name: Find a field labeled "username" in an item named "MySQL Database" in a specific vault.
  onepassword.connect.field_info:
    item: MySQL Database
    field: username
    vault: 2zbeu4rexampleuxmyvhdh57b6

- name: Find a field labeled "username" in a specific section.
  onepassword.connect.field_info:
    item: MySQL Database
    section: Credentials
    field: username
    vault: 2zbeu4rexampleuxmyvhdh57b6
```

The playbook receives the field data with this shape:
```
{
    "field": { 
        "value": str, 
        "section": str|null, 
        "id": str 
    }
}
```

### CODE REVIEW

File: plugins/module_utils/api.py [hunk 1]
```diff
@@ -189,3 +192,23 @@ def _format_user_agent(collection_version, python_version=None, ansible_version=
...
         ansible=ansible_version or "unknown"
     )
+
+
+# Client UUIDs must be exactly 26 characters.
+CLIENT_UUID_LENGTH = 26
+
+
+def valid_client_uuid(uuid):
+    if len(uuid) > CLIENT_UUID_LENGTH:
```
Comment (hculea) [inline plugins/module_utils/api.py:202]: Could you please explain why do we allow uuid to be shorter, and we only check for whether it is longer?
Does the below regex match take care of that?

File: plugins/module_utils/api.py [hunk 2]
```diff
@@ -189,3 +192,23 @@ def _format_user_agent(collection_version, python_version=None, ansible_version=
...
+        return False
+    # triple curly braces needed to escape f-strings as regex quantifiers
+    return re.match(rf"^[0-9a-z]{{{CLIENT_UUID_LENGTH}}}$", uuid) is not None
+
+
+def create_client_uuid():
+    """Create a valid client UUID.
+
+    The UUID does is not intended to be cryptographically random."""
```
Comment (hculea) [inline plugins/module_utils/api.py:211]: I think there's a typo here.

File: tests/integration/targets/field_info/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+####################################################################
+# WARNING: These are designed specifically for Ansible tests       #
+# and should not be used as examples of how to write Ansible roles #
+####################################################################
+---
+- import_tasks: tests.yml
+  environment:
+    OP_CONNECT_HOST: "__OP_CONNECT_HOST__"
+    OP_CONNECT_TOKEN: "__OP_CONNECT_TOKEN__"
+    OP_VAULT_ID: "__OP_VAULT_ID__"
```
Comment (hculea) [inline tests/integration/targets/field_info/tasks/main.yml:10]: Let's add a newline here, for consistency.

File: tests/integration/targets/field_info/tasks/tests.yml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+- name: Find Field by Section | Assert field within MySQL section was found
+  assert:
+    that:
+      - mysql_field.field.value == "penny"
+      - mysql_field.field.section
+
+- name: Cleanup | Remove test items
+  generic_item:
+    state: absent
+    uuid: "{{ test_item.op_item.id }}"
```
Comment (hculea) [inline tests/integration/targets/field_info/tasks/tests.yml:80]: And here.

File: plugins/module_utils/specs.py [hunk 5]
```diff
@@ -68,8 +68,15 @@ def op_item_info():
             type="str",
             required=True
         ),
+        flatten_fields_by_label=dict(
+            type="bool",
+            default=True
```
Comment (hculea) [inline plugins/module_utils/specs.py:73]: even if the default behaviour will be returning the fields as a list starting from 3.0.0, we will still be allowing users to set `flatten_fields_by_label` to `true`, in order to get a dictionary if this is the desired behaviour, right?

**Review by hculea — APPROVED** [review]
Good job!

It's been amazing getting onboarded on this repo by exploring this PR. Let's ship!!

File: plugins/module_utils/api.py (hunk 1 above)
Comment (verkaufer) [reply plugins/module_utils/api.py:202]: Yes, the regex below will ensure the string is exactly 26 characters long. I see your point, though, so I'll get rid of the check here and just rely on the regex.

Confirmation: https://regex101.com/r/NrLJT6/1

File: plugins/module_utils/specs.py (hunk 5 above)
Comment (verkaufer) [reply plugins/module_utils/specs.py:73]: Good point! I agree we should keep the option around in v3.0.0 and instead switch the default to `default=False`

**Pull request merged**